packages/ui/src/components/Generic/InputPopUp/InputPopUp.tsx

```diff
@@ -0,0 +1,151 @@
+import { Button, Heading, TextInput } from "@repo/ui/components/Primitive"
+import {
+  Center,
+  Dialog,
+  DialogBody,
+  DialogCloseButton,
+  DialogFooter,
+  DialogHeader,
+  DialogOverlay,
+  ErrorMessage,
+  FormControl,
+  Stack,
+  // useDisclosure,
+} from "@yamada-ui/react"
+import { type SubmitHandler, useForm } from "react-hook-form"
+
+/**
+ * Props for {@link InputPopUp} component
+ */
+interface FormData {
+  name: string
+}
+export interface InputPopUpProps {
+  /**
+   * The title to display in the dialog header
+   */
+  title: string
+  /**
+   * The open state of the dialog
+   */
+  open: boolean
+  /**
+   * The function to call when the dialog is closed
+   */
+  onClose: () => void
+  /**
+   * The description text to display in the dialog body
+   */
+  description: string
+  /**
+   * The function to call when the form is submitted
+   */
+  onConfirm: (value: string) => void
+}
+
+/**
+ * Dialog component for collecting user input with a title and description
+ *
+ * Renders a modal dialog with a title heading, description text, and an input field,
+ * designed to collect user input in a clean, accessible interface.
+ *
+ * @param props InputPopUp component properties
+ * @returns A modal dialog with title, description, and input field
+ *
+ * @example
+<InputPopUp
+        description="Enter Semester Name:"
+        onClose={onClose}
+        onConfirm={(val) => console.log("Confirmed:", val)}
+        open={open}
+        title="Create New Semester"
+      />
+ */
+export const InputPopUp: React.FC<InputPopUpProps> = ({
+  open,
+  onClose,
+  title,
+  description,
+  onConfirm,
+}) => {
+  // const { open, onOpen, onClose } = useDisclosure()
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm<FormData>()
+  const onSubmit: SubmitHandler<FormData> = (data) => {
+    onConfirm(data.name)
+    reset()
+    onClose()
+  }
+  // const { open, onOpen, onClose } = useDisclosure()
+  return (
+    <>
+      {/* <Button onClick={onOpen}>{buttonText}</Button> */}
+
+      <Dialog
+        alignItems="center"
+        as="form"
+        bg={["secondary.50", "secondary.900"]}
+        // backdropBlur="15px"
+        boxShadow="0px 1.5px 0px 0px rgba(0, 0, 0, 0.05), 0px 6px 6px 0px rgba(0, 0, 0, 0.05), 0px 15px 15px 0px rgba(0, 0, 0, 0.1)"
+        gap={{ base: "6", md: "13" }}
+        justifyContent="center"
+        layerStyle="gradientBorder"
+        onClose={onClose}
+        onSubmit={handleSubmit(onSubmit)}
+        open={open}
+        p={{ base: "6", md: "13" }}
+        rounded="2xl"
+        size="3xl"
+      >
+        <DialogOverlay />
+        <DialogCloseButton bg="black" layerStyle="gradientBorder" rounded="full" size="sm" />
+        {/* <form noValidate onSubmit={handleSubmit(onSubmit)}> */}
```

Comment:
I think considering the title and description is very big in this component, the close button should be `size='md'` (default).
```suggestion
      >
        <DialogOverlay />
        <DialogCloseButton bg="black" layerStyle="gradientBorder" rounded="full" />
```